src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2002,6 +2002,20 @@ private BoundExpression BindIncrementOperator(CSharpSyntaxNode node, ExpressionS
                 hasErrors);
         }
 
+        private BoundExpression BindNotNullExpression(PostfixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            var expr = BindValue(node.Operand, diagnostics, BindValueKind.RValue);
```

Comment:
We probably should bind in the same manner BindParenthesizedExpression binds underlying expression and then adjust CheckValue (and the like) to understand the "pass-through" nature of BoundNotNullExpression.